### PR TITLE
exiv2: 0.28.0 -> 0.28.1

### PR DESCRIPTION
**Pull request by 6t8k**
## Description of changes

https://github.com/Exiv2/exiv2/blob/v0.28.1/doc/ChangeLog

Fixes [CVE-2023-44398](https://github.com/Exiv2/exiv2/security/advisories/GHSA-hrw9-ggg3-3r4r) (High).

https://github.com/Exiv2/exiv2/issues/2762 is now fixed upstream, so make `checkPhase` run unconditionally again.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Just add `"-DCMAKE_INSTALL_DATADIR=${placeholder "dev"}/share"` to `cmakeFlags`.

**Review by wegank — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Comment by risicle** [comment]
@ofborg build photoqt

**Comment by risicle** [comment]
@ofborg build digikam

**Comment by wegank** [comment]
Yes, this PR needs to provide patches for `digikam`, `kphotoalbum` and `photoqt`.

**Comment by wegank** [comment]
https://invent.kde.org/graphics/digikam/-/commit/f5ea91a7f6c1926815ec68f3e0176d6c15b83051 could probably help.

File: pkgs/applications/graphics/photoqt/default.nix [hunk 1]
```diff
@@ -30,8 +30,13 @@ stdenv.mkDerivation rec {
     hash = "sha256-kVf9+zI9rtEMmS0N4qrN673T/1fnqfcV3hQPnMXMLas=";
   };
 
+  postPatch = ''
```
Comment (wegank) [inline pkgs/applications/graphics/photoqt/default.nix:33]: That's basically https://gitlab.com/lspies/photoqt/-/commit/23e03360adefebadbccfb4ac4f1628caeea216f7. Does the patch apply cleanly here?

Comment (wegank) [reply pkgs/applications/graphics/photoqt/default.nix:33]: Oh, didn't see your comment.

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
No more regression on x86_64-linux.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 266341` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages marked as broken and skipped:</summary>
  <ul>
    <li>cataract</li>
    <li>cataract-unstable</li>
    <li>entangle</li>
    <li>gimp-with-plugins</li>
    <li>gnome.nautilus-python</li>
    <li>gnome.nautilus-python.dev</li>
    <li>gnome.nautilus-python.devdoc</li>
    <li>gnome.nautilus-python.doc</li>
    <li>kphotoalbum</li>
    <li>krename</li>
    <li>latte-dock</li>
    <li>libsForQt5.arianna</li>
    <li>libsForQt5.bismuth</li>
    <li>libsForQt5.krohnkite</li>
    <li>libsForQt5.kwin-dynamic-workspaces</li>
    <li>libsForQt5.kwin-tiling</li>
    <li>libsForQt5.kzones</li>
    <li>libsForQt5.parachute</li>
    <li>lightly-boehs</li>
    <li>lightly-qt</li>
    <li>nordic</li>
    <li>nordic.sddm</li>
    <li>plasma5Packages.arianna</li>
    <li>plasma5Packages.bismuth</li>
    <li>plasma5Packages.krohnkite</li>
    <li>plasma5Packages.kwin-dynamic-workspaces</li>
    <li>plasma5Packages.kwin-tiling</li>
    <li>plasma5Packages.kzones</li>
    <li>plasma5Packages.parachute</li>
    <li>python310Packages.py3exiv2</li>
    <li>python310Packages.py3exiv2.dist</li>
    <li>python311Packages.py3exiv2</li>
    <li>python311Packages.py3exiv2.dist</li>
    <li>toppler</li>
    <li>variety</li>
    <li>variety.dist</li>
    <li>vimiv-qt</li>
    <li>vimiv-qt.dist</li>
  </ul>
</details>
<details>
  <summary>29 packages built:</summary>
  <ul>
    <li>darktable</li>
    <li>exiv2</li>
    <li>exiv2.dev</li>
    <li>exiv2.doc</li>
    <li>exiv2.lib</li>
    <li>exiv2.man</li>
    <li>gegl</li>
    <li>gegl.dev</li>
    <li>gegl.devdoc</li>
    <li>gexiv2</li>
    <li>gexiv2.dev</li>
    <li>gexiv2.devdoc</li>
    <li>gimp (gimpPlugins.gimp)</li>
    <li>gimp.dev (gimpPlugins.gimp.dev)</li>
    <li>gimpPlugins.gmic</li>
    <li>gnunet</li>
    <li>gnunet-gtk</li>
    <li>gpscorrelate</li>
    <li>gramps</li>
    <li>gramps.dist</li>
    <li>libextractor</li>
    <li>libsForQt5.kfilemetadata (plasma5Packages.kfilemetadata)</li>
    <li>libsForQt5.kfilemetadata.bin (plasma5Packages.kfilemetadata.bin)</li>
    <li>libsForQt5.kfilemetadata.dev (plasma5Packages.kfilemetadata.dev)</li>
    <li>merkaartor</li>
    <li>nomacs</li>
    <li>photoqt</li>
    <li>tracker-miners</li>
    <li>viking</li>
  </ul>
</details>

**Pull request merged**